### PR TITLE
Improve validations for student

### PR DESCRIPTION
**Pull request by holenet**
Based on the [Student API](#11),

Added validations:
- email: host must be 'snu.ac.kr'
- grade: must be 1~4
- department: must belong to the college
- major: must belong to the department

& minor improvements

### CODE REVIEW

File: backend/ttrs/serializers.py [hunk 1]
```diff
@@ -2,23 +2,60 @@
...
 class StudentSerializer(serializers.ModelSerializer):
     class Meta:
         model = Student
-        fields = ('id', 'username', 'password', 'email', 'grade', 'college', 'department', 'major')
+        fields = ('id', 'username', 'password', 'email', 'grade', 'college', 'department', 'major', 'not_recommends')
         extra_kwargs = {
             'email': {'required': True, 'allow_null': False, 'allow_blank': False}
         }
 
+    def validate_email(self, value):
```
Comment (lameinthebox) [inline backend/ttrs/serializers.py]: Would you rename the parameter from _value_ to  _email_?

File: backend/ttrs/serializers.py [hunk 2]
```diff
@@ -2,23 +2,60 @@
...
+        if value.split('@')[1] != 'snu.ac.kr':
+            raise ValidationError("The host must be 'snu.ac.kr'.")
+        return value
+
+    def validate_grade(self, value):
```
Comment (lameinthebox) [inline backend/ttrs/serializers.py]: Here too. _value_ to _grade_?

File: backend/ttrs/serializers.py [hunk 3]
```diff
@@ -2,23 +2,60 @@
...
     def validate(self, data):
+        # belonging validations of college-department, department-major
+        errors = []
+        college = self.get_field_value(data, 'college')
+        department = self.get_field_value(data, 'department')
+        major = self.get_field_value(data, 'major')
+        if department is not None:
+            if college and department and college.id != department.college_id:
+                errors.append('The department must belong to the college')
+            if major is not None:
```
Comment (lameinthebox) [inline backend/ttrs/serializers.py]: I think it would be better if you tidy _nested if_.

Comment (holenet) [reply backend/ttrs/serializers.py]: @lameinthebox 
I don't understand your intent.
Could you write an example?

Comment (holenet) [reply backend/ttrs/serializers.py]: @lameinthebox 
Sorry, I got your point.
I'll fix it.

**Pull request merged**